### DISCUSSION TITLE
Fullsize App for Linux (Skia)

### DISCUSSION THREAD

**Discussion by pkarsten** [unoplatform/uno#16954] [General]
Hi, 
I would like to port my C# UWP app to Linux for a Raspberry Pi. I want the app to run in full screen, but I can't get it to work. I have created a Uno app with the following code. 

`dotnet new unoapp -o PiPic -preset "recommended" -platforms "android" -platforms "windows" -platforms "desktop" -auth "custom" -theme-service False -vscode  -pwa False -maui False -theme "fluent" -presentation "mvvm" -config False -di  -log "default" -nav "regions" -http False -loc  -tests "none" -toolkit  -dsp False -id "com.companyname.PiPic" -pub "O=PiPic"`

I then tried to modify the Program.cs by adding the Uno.UI.Runtime.Skia, but this leads to a bunch of compile errors, Uno.SourceGenerator . xaml generator error

```csharp
using Uno.UI.Runtime.Skia;
using Gtk;
using Window = Gtk.Window;

namespace PiPic;

public class Program
{
    [STAThread]
    public static void Main(string[] args)
    {

        var host = SkiaHostBuilder.Create()
                    .App(() => new App())
                    .UseX11()
                    .UseLinuxFrameBuffer()
                    .UseMacOS()
                    .UseWindows()
                    .Build();

        host.Run();

        // Access the main GTK window and configure it
        var windows = Window.ListToplevels();
        if (windows.Length > 0 && windows[0] is Window window)
        {
            // Remove the title bar
            window.Decorated = false;

            // Set the window to full screen
            window.Fullscreen();
        }

        Gtk.Application.Run();
    }
}
```

Is there an “easy” way to run the app in full screen? I would like to program the app for Windows, Linux (Raspoberry Pi) and Android. Thank you 

Regards 
Peter

**Comment by jeromelaban** [comment]
Thanks for the report. Could you provide a [binlog the issue](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) so we can troubleshoot the build error?

For fullscreen, you can either use the linux framebuffer, or you can use the [fullscreen APIs](https://platform.uno/docs/articles/features/windows-ui-xaml-window.html#making-the-window-full-screen).
